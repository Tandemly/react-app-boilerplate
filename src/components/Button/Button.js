// @flow
import * as React from "react";
import cx from "classnames";
import styleable from "react-styleable";
import styles from "./styles.module.scss";

type Props = {
  /** the text label on the button */
  children?: React.Node,
  /** an alternate way to pass a label to display on the button (overrides `children`) */
  text?: string,
  /** classnames to add to the rendered component */
  className?: string,
  /** style object from react-styleable */
  css?: { [key: string]: any }
};

/**
 * A basic button component.
 */
const Button = ({ children, text, className, css }: Props) => (
  <button className={cx("button", css.button, className)}>
    { (text || children) && <span>{text || children}</span> }
  </button>
);

export default styleable(styles)(Button);
