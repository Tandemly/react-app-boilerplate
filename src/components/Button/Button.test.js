import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import Button from "./Button";

it("renders without crashing", () => {
  const button = shallow(<Button />);
  expect(button).toMatchSnapshot();
});

it("renders with `text` prop and `children`", () => {
  const text = "Test";
  let button = mount(<Button>{text}</Button>);
  expect(button.text()).toBe(text);

  button = mount(<Button text={text} />);
  expect(button.text()).toBe(text);
});

it("renders with proper classes/styles", () => {
  const text = "Test";
  let button = mount(<Button className="other">{text}</Button>);
  expect(button.find("button").hasClass("button")).toBeTruthy(); // bulma specific
  expect(button.find("button").hasClass("other")).toBeTruthy(); // via props
});
