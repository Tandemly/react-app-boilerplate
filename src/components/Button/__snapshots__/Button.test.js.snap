// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styleable(Button)
    css={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "css": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "css": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
